tests/utils/mod.rs

```diff
@@ -0,0 +1,264 @@
+//! # Test utils
+//!
+//! Most tests require to set up groups, clients, credentials, and identities.
+//! This module implements helpers to do that.
+//!
+
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+use evercrypt::prelude::*;
+use openmls::prelude::*;
+use rand::rngs::OsRng;
+use rand::RngCore;
+
+/// Configuration of a client meant to be used in a test setup.
+#[derive(Clone)]
+pub(crate) struct TestClientConfig {
+    /// Name of the client.
+    pub(crate) name: &'static str,
+    /// Ciphersuites supported by the client.
+    pub(crate) ciphersuites: Vec<CiphersuiteName>,
+}
+
+/// Configuration of a group meant to be used in a test setup.
+pub(crate) struct TestGroupConfig {
+    pub(crate) ciphersuite: CiphersuiteName,
+    pub(crate) config: GroupConfig,
+    pub(crate) members: Vec<TestClientConfig>,
+}
+
+/// Configuration of a test setup including clients and groups used in the test
+/// setup.
+pub(crate) struct TestSetupConfig {
+    pub(crate) clients: Vec<TestClientConfig>,
+    pub(crate) groups: Vec<TestGroupConfig>,
+}
+
+/// A client in a test setup.
+pub(crate) struct TestClient {
+    pub(crate) credential_bundles: HashMap<CiphersuiteName, CredentialBundle>,
+    pub(crate) key_package_bundles: RefCell<Vec<KeyPackageBundle>>,
+    pub(crate) group_states: RefCell<HashMap<GroupId, MlsGroup>>,
+}
+
+/// The state of a test setup, including the state of the clients and the
+/// keystore, which holds the KeyPackages published by the clients.
+pub(crate) struct TestSetup {
+    pub(crate) key_store: RefCell<HashMap<(&'static str, CiphersuiteName), Vec<KeyPackage>>>,
```

Comment:
This seems unused right now.

Reply:
It is right now, but in future tests, the tester will probably want to access the `KeyPackage`s published by the parties. So it should be used in the future.

Reply:
Sure, that's fine. But it triggers a clippy warning so you should at least prefix it with an underscore.